libcaf_core/caf/type_nr.hpp

```diff
@@ -24,65 +24,242 @@
+  constexpr name name##_v = name {                                             \
+  }
+
 namespace caf {
 
+/// Signals the receiver to add the operands.
+CAF_MSG_TYPE_ADD_ATOM(add_atom);
+
+/// Signals the receiver to subtract the operands.
+CAF_MSG_TYPE_ADD_ATOM(sub_atom);
+
+/// Signals the receiver to multiply the operands.
+CAF_MSG_TYPE_ADD_ATOM(mul_atom);
+
+/// Signals the receiver to divide the operands.
+CAF_MSG_TYPE_ADD_ATOM(div_atom);
+
+/// Used for request operations.
+CAF_MSG_TYPE_ADD_ATOM(get_atom);
+
+/// Used for request operations.
+CAF_MSG_TYPE_ADD_ATOM(put_atom);
+
+/// Used for signalizing resolved paths.
+CAF_MSG_TYPE_ADD_ATOM(resolve_atom);
+
+/// Used for signalizing updates, e.g., in a key-value store.
+CAF_MSG_TYPE_ADD_ATOM(update_atom);
+
+/// Used for request operations.
+CAF_MSG_TYPE_ADD_ATOM(delete_atom);
+
+/// Used for response messages.
+CAF_MSG_TYPE_ADD_ATOM(ok_atom);
+
+/// Used for triggering system-level message handling.
+CAF_MSG_TYPE_ADD_ATOM(sys_atom);
+
+/// Used for signaling group subscriptions.
+CAF_MSG_TYPE_ADD_ATOM(join_atom);
+
+/// Used for signaling group unsubscriptions.
+CAF_MSG_TYPE_ADD_ATOM(leave_atom);
+
+/// Used for signaling forwarding paths.
+CAF_MSG_TYPE_ADD_ATOM(forward_atom);
+
+/// Used for buffer management.
+CAF_MSG_TYPE_ADD_ATOM(flush_atom);
+
+/// Used for I/O redirection.
+CAF_MSG_TYPE_ADD_ATOM(redirect_atom);
+
+/// Used for link requests over network.
+CAF_MSG_TYPE_ADD_ATOM(link_atom);
+
+/// Used for removing networked links.
+CAF_MSG_TYPE_ADD_ATOM(unlink_atom);
+
+/// Used for monitor requests over network.
+CAF_MSG_TYPE_ADD_ATOM(monitor_atom);
+
+/// Used for removing networked monitors.
+CAF_MSG_TYPE_ADD_ATOM(demonitor_atom);
+
+/// Used for publishing actors at a given port.
+CAF_MSG_TYPE_ADD_ATOM(publish_atom);
+
+/// Used for publishing actors at a given port.
+CAF_MSG_TYPE_ADD_ATOM(publish_udp_atom);
+
+/// Used for removing an actor/port mapping.
+CAF_MSG_TYPE_ADD_ATOM(unpublish_atom);
+
+/// Used for removing an actor/port mapping.
+CAF_MSG_TYPE_ADD_ATOM(unpublish_udp_atom);
+
+/// Used for signalizing group membership.
+CAF_MSG_TYPE_ADD_ATOM(subscribe_atom);
+
+/// Used for withdrawing group membership.
+CAF_MSG_TYPE_ADD_ATOM(unsubscribe_atom);
+
+/// Used for establishing network connections.
+CAF_MSG_TYPE_ADD_ATOM(connect_atom);
+
+/// Used for contacting a remote UDP endpoint
+CAF_MSG_TYPE_ADD_ATOM(contact_atom);
+
+/// Used for opening ports or files.
+CAF_MSG_TYPE_ADD_ATOM(open_atom);
+
+/// Used for closing ports or files.
+CAF_MSG_TYPE_ADD_ATOM(close_atom);
+
+/// Used for spawning remote actors.
+CAF_MSG_TYPE_ADD_ATOM(spawn_atom);
+
+/// Used for migrating actors to other nodes.
+CAF_MSG_TYPE_ADD_ATOM(migrate_atom);
+
+/// Used for triggering periodic operations.
+CAF_MSG_TYPE_ADD_ATOM(tick_atom);
+
+/// Used for pending out of order messages.
+CAF_MSG_TYPE_ADD_ATOM(pending_atom);
+
+/// Used as timeout type for `timeout_msg`.
```

Comment:
Yeah.... Captain obvious wants his comment back. :)

Since we remove them anyway (see new type ID API), I think fixing this is a waste of time, tho.